resources/run_adl.sh

```diff
@@ -146,25 +163,35 @@ check_mongod
 if [[ $? -ne 0 ]]; then
   if [ $ARG = $START ]; then
     echo "Starting $MONGOD"
-    # Install and start mongod
-    if [ $OS = "Linux" ]; then
-      (cd $LOCAL_INSTALL_DIR && curl -O $MONGO_DOWNLOAD_BASE/linux/$MONGO_DOWNLOAD_LINK)
-    else
-      (cd $LOCAL_INSTALL_DIR && curl -O $MONGO_DOWNLOAD_BASE/osx/$MONGO_DOWNLOAD_LINK)
-    fi
-
-    # Uncompressed the archive
-    tar zxvf $LOCAL_INSTALL_DIR/$MONGO_DOWNLOAD_LINK --directory $LOCAL_INSTALL_DIR
-    MONGO_DOWNLOAD_DIR=$LOCAL_INSTALL_DIR/${MONGO_DOWNLOAD_LINK:0:$((${#MONGO_DOWNLOAD_LINK} - 4))}
 
     mkdir -p $MONGO_DB_PATH
     mkdir -p $LOGS_PATH
     mkdir -p $TMP_DIR
 
-    # Note: ADL has a storage.json file that generates configs for us.
-    # The mongodb source is on port $MONGOD_PORT so we use that here.
-    $MONGO_DOWNLOAD_DIR/bin/mongod --port $MONGOD_PORT --dbpath $MONGO_DB_PATH \
-      --logpath $LOGS_PATH/mongodb_test.log --pidfilepath $TMP_DIR/${MONGOD}.pid --fork
+    # Install and start mongod
+    (cd $LOCAL_INSTALL_DIR && curl -O $MONGO_DOWNLOAD_LINK)
+
+    # Uncompress the archive
+    if [[ $OS =~ ^CYGWIN ]]; then
+      unzip -o $LOCAL_INSTALL_DIR/$MONGO_DOWNLOAD_FILE -d $LOCAL_INSTALL_DIR
+
+      # Obtain unzipped directory name
+      MONGO_UNZIP_DIR=$(unzip -l $LOCAL_INSTALL_DIR/$MONGO_DOWNLOAD_FILE | grep mongod.exe | tr -s ' ' \
+	          | cut -d ' ' -f 5 | cut -d/ -f1)
+      chmod +x $LOCAL_INSTALL_DIR/$MONGO_UNZIP_DIR/bin/mongod.exe
+      MONGO_DOWNLOAD_DIR=$LOCAL_INSTALL_DIR/$MONGO_UNZIP_DIR
+      # mongod does not have --fork option on Windows, using nohup
+      nohup $MONGO_DOWNLOAD_DIR/bin/mongod --port $MONGOD_PORT --dbpath $(cygpath -m ${MONGO_DB_PATH}) \
+              --logpath $(cygpath -m $LOGS_PATH/mongodb_test.log) &
+      echo $! > $TMP_DIR/${MONGOD}.pid
+    else
+      tar zxvf $LOCAL_INSTALL_DIR/$MONGO_DOWNLOAD_FILE --directory $LOCAL_INSTALL_DIR
+      # Note: ADL has a storage.json file that generates configs for us.
```

Comment:
Do you want to move this comment up, above the if? It seems to apply to both section.

Reply:
Sounds good, I moved it up.